src/zos_tso/zowe/zos_tso_for_zowe_sdk/tso.py

```diff
@@ -61,6 +60,11 @@ def issue_command(self, command):
         session_key = self.start_tso_session()
         command_output = self.send_tso_message(session_key, command)
         tso_messages = self.retrieve_tso_messages(command_output)
+        while not any("TSO PROMPT" in message for message in command_output) or not tso_messages:
+           custom_args = self._create_custom_request_arguments()
+           custom_args["url"] = "{}/{}".format(self.request_endpoint, session_key)
+           command_output = self.request_handler.perform_request("GET", custom_args)["tsoData"]
```

Comment:
The code for getting TSO data could be moved into a separate helper method and invoked like this:
```suggestion
           command_output = self.get_tso_data(session_key)
```